File: internal/model/product.go
package model

const (
	ProductStatusActive   = 1
	ProductStatusInactive = 0
)

type Product struct {
	BaseModel
	SKU           string  `bun:"sku,notnull,unique" json:"sku"`
	Name          string  `bun:"name,notnull" json:"name"`
	CategoryID    int64   `bun:"category_id" json:"category_id"`
	Specification string  `bun:"specification" json:"specification"`
	Unit          string  `bun:"unit" json:"unit"`
	Barcode       string  `bun:"barcode" json:"barcode"`
	Price         float64 `bun:"price" json:"price"`
	Description   string  `bun:"description" json:"description"`
	Status        int     `bun:"status,notnull" json:"status"`
}

func (p *Product) TableName() string {
	return "products"
}

func (p *Product) IsActive() bool {
	return p.Status == ProductStatusActive
}

File: web/src/types/product.ts
export interface Category {
  id: number
  name: string
  parent_id: number | null
  sort_order: number
  status: number
  created_at: string
  updated_at: string
  children?: Category[]
}

export interface CreateCategoryRequest {
  name: string
  parent_id?: number
  sort_order?: number
  status?: number
}

export interface UpdateCategoryRequest {
  name?: string
  parent_id?: number
  sort_order?: number
  status?: number
}

export interface Product {
  id: number
  sku: string
  name: string
  category_id: number | null
  specification: string
  unit: string
  barcode: string
  price: number
  description: string
  status: number
  created_at: string
  updated_at: string
}

export interface CreateProductRequest {
  sku: string
  name: string
  category_id?: number
  specification?: string
  unit?: string
  barcode?: string
  price?: number
  description?: string
  status?: number
}

export interface UpdateProductRequest {
  sku?: string
  name?: string
  category_id?: number
  specification?: string
  unit?: string
  barcode?: string
  price?: number
  description?: string
  status?: number
}
